contrib/net/http/option.go

```diff
@@ -84,65 +54,55 @@ func WithServiceName(name string) MuxOption {
 // Special headers can not be sub-selected. E.g., an entire Cookie header would be transmitted, without the ability to choose specific Cookies.
 func WithHeaderTags(headers []string) Option {
 	headerTagsMap := normalizer.HeaderTagSlice(headers)
-	return func(cfg *config) {
-		cfg.headerTags = internal.NewLockMap(headerTagsMap)
-	}
-}
-
-// WithStatusCheck sets a span to be an error if the passed function
-// returns true for a given status code.
-func WithStatusCheck(fn func(statusCode int) bool) Option {
-	return func(cfg *config) {
-		cfg.isStatusError = fn
+	return func(cfg *cfg.Config) {
+		cfg.HeaderTags = internal.NewLockMap(headerTagsMap)
 	}
 }
 
 // WithAnalytics enables Trace Analytics for all started spans.
 func WithAnalytics(on bool) MuxOption {
-	return func(cfg *config) {
+	return func(cfg *cfg.Config) {
 		if on {
-			cfg.analyticsRate = 1.0
-			cfg.spanOpts = append(cfg.spanOpts, tracer.Tag(ext.EventSampleRate, cfg.analyticsRate))
+			cfg.AnalyticsRate = 1.0
+			cfg.SpanOpts = append(cfg.SpanOpts, tracer.Tag(ext.EventSampleRate, cfg.AnalyticsRate))
 		} else {
-			cfg.analyticsRate = math.NaN()
+			cfg.AnalyticsRate = math.NaN()
 		}
 	}
 }
 
 // WithAnalyticsRate sets the sampling rate for Trace Analytics events
 // correlated to started spans.
-func WithAnalyticsRate(rate float64) MuxOption {
-	return func(cfg *config) {
+func WithAnalyticsRate(rate float64) Option {
+	return func(cfg *cfg.Config) {
```

Comment:
`cfg` is a very common variable name and this kind of shadowing might be error prone / confusing. WDYT about changing the import alias to something like `internalconfig`?